worker_tasks/categorize.py

```diff
@@ -1,11 +1,44 @@
-from collections import Counter
+# categorize.py
+# 
+# Adapted from algorithm by Grineva, et. al. presented at:
+# http://www.slideshare.net/maria.grineva/extracting-key-terms-from-noisy-and-multitheme-documents?related=1
+# 
+# Algorithm Overview
+# 1) Get initial list of possible keyword candidates using ngram frequencies
+# 2) For each keyword candidate, use Wikipedia to extract all related article titles
+#       a) Related articles are called (disambiguation links) on Wikipedia
+# 3) Try to assign one Wikipedia article to each keyword candidate
+#       a) First, assign articles to the keywords with only 1 related article
+#       b) Use these "auto-assigned" articles as the context to narrow the other categories' article
+#       c) With the context from (b), compute relatedness scores to find the best article match
+#           i) relatedness score = [SUM OF SHARED WIKIPEDIA HYPERLINKS] / [TOTAL NUMBER OF WIKIPEDIA HYPERLINKS]
+# 4) With each keyword assigned 1 article, create a graph where each vertex is an article and the edge weight
```

Comment:
Can a keyword be assigned more than 1 article by this step? Or can it be assigned zero?